pinot-controller/src/test/java/org/apache/pinot/controller/ControllerConfTest.java

```diff
@@ -0,0 +1,193 @@
+
+  private static final List<String> NEW_CONFIGS = Arrays
+      .asList(RETENTION_MANAGER_FREQUENCY_PERIOD, OFFLINE_SEGMENT_INTERVAL_CHECKER_FREQUENCY_PERIOD,
+          REALTIME_SEGMENT_VALIDATION_FREQUENCY_PERIOD, BROKER_RESOURCE_VALIDATION_FREQUENCY_PERIOD,
+          STATUS_CHECKER_FREQUENCY_PERIOD, TASK_MANAGER_FREQUENCY_PERIOD,
+          MINION_INSTANCES_CLEANUP_TASK_FREQUENCY_PERIOD,
+          MINION_INSTANCES_CLEANUP_TASK_MIN_OFFLINE_TIME_BEFORE_DELETION_PERIOD, TASK_METRICS_EMITTER_FREQUENCY_PERIOD,
+          SEGMENT_RELOCATOR_FREQUENCY_PERIOD, SEGMENT_LEVEL_VALIDATION_INTERVAL_PERIOD,
+          STATUS_CHECKER_WAIT_FOR_PUSH_TIME_PERIOD);
+
+  private static final Random RAND = new Random();
+
+  /**
+   * When config contains: 1. Both deprecated config and the corresponding new config. 2. All new
+   * configurations are valid. 3. Some deprecated configurations are invalid. then new configs
+   * override deprecated configs (invalid deprecated configs do not throw exceptions when
+   * corresponding valid new configs are supplied as well)
+   */
+  @Test
+  public void validNewConfigOverridesCorrespondingValidOrInvalidOldConfigOnRead() {
+    //setup
+    Map<String, Object> controllerConfig = new HashMap<>();
+    int durationInSeconds = getRandomDurationInSeconds();
+    DEPRECATED_CONFIGS.forEach(config -> controllerConfig.put(config, durationInSeconds));
+    //put some invalid deprecated configs
+    controllerConfig.put(DEPRECATED_RETENTION_MANAGER_FREQUENCY_IN_SECONDS, getRandomString());
+    controllerConfig.put(DEPRECATED_SEGMENT_LEVEL_VALIDATION_INTERVAL_IN_SECONDS, getRandomString());
+    //override all deprecated configs with valid new configs
+    String period = getRandomPeriodInMinutes();
+    NEW_CONFIGS.forEach(config -> controllerConfig.put(config, period));
+    ControllerConf conf = new ControllerConf(controllerConfig);
+    //execution and assertion
+    assertOnDurations(conf, TimeUnit.SECONDS.convert(TimeUtils.convertPeriodToMillis(period), TimeUnit.MILLISECONDS));
+  }
+
+  /**
+   * When config contains: 1. Both deprecated config and the corresponding new config. 2. All
+   * deprecated configurations are valid. 3. Some new configurations are invalid. then exceptions
+   * are thrown when invalid new configurations are read (there is no fall-back to the corresponding
+   * valid deprecated configuration). For all valid new configurations, they override the
+   * corresponding deprecated configuration.
+   */
+  @Test
+  public void invalidNewConfigShouldThrowExceptionOnReadWithoutFallbackToCorrespondingValidDeprecatedConfig() {
+    //setup
+    Map<String, Object> controllerConfig = new HashMap<>();
+    int durationInSeconds = getRandomDurationInSeconds();
+    //all deprecated configs should be valid
+    DEPRECATED_CONFIGS.forEach(config -> controllerConfig.put(config, durationInSeconds));
+    String randomPeriodInMinutes = getRandomPeriodInMinutes();
```

Comment:
In this case maybe it is ok, but in general, it is a good idea to set the seed for the random number up front, and print out the value of the seed (in system.out). This will make sure that if there is failure due to some random value, we can reproduce the test case by manually setting the seed to that value. Otherwise, we will face intermittent failures in tests without knowing why that happened.

Reply:
Ah this is a good learning. Will incorporate this change.

Reply:
While this definitely is needed for reproducing failing tests, in this case, simply printing the test configuration gives enough information to reproduce a failing test. So I have simply printing the config for any failing test.